data/api/server-server/user_keys.yaml

```diff
@@ -71,26 +71,28 @@ paths:
                         type: object
                         # Key
                         additionalProperties:
-                          type: object
-                          title: KeyObject
-                          properties:
-                            key:
-                              type: string
-                              description: The key, encoded using unpadded base64.
-                            signatures:
-                              type: object
-                              title: Signatures
-                              additionalProperties:
-                                type: object
-                                additionalProperties:
+                          oneOf:
+                            - type: string
+                            - type: object
```

Comment:
Good question - might very well be a copy-pasta on my end. Can one-time keys be unsigned - guess not? If not, my suggestion is incorrect and there should be `KeyObject` here, without alternatives.

Reply:
Ok, so it's not a copy-pasta on my end at least. The alternative between `string` and `object` has been here [since the very beginning](https://github.com/zecakeh/matrix-spec/commit/80699812525655fa1f251eb527993af3c71df955#diff-817d65ea24375c8feef6f013fdbf5ccb256dec98585df0a4dac07752651b79f0R77), even though undocumented. My comment in #1351 referred to the correct nesting level and I mentioned that `string` has to be returned only because it has been there before. If it hasn't been correct from the beginning then I'd rather we fix it in a separate PR but I'm fine either way.

Reply:
Right, I think I've figured this out!

Before we invented the `signed_curve25519` key algorithm for one-time-keys, we just used `curve25519` keys. These, not being signed, can be represented just by a string.

The `curve25519` algorithm should no longer be used here, but removing them from the spec sounds like a different job.

So, :+1: to this fix.

Reply:
(more on algorithms at https://spec.matrix.org/v1.7/client-server-api/#key-algorithms and https://gitlab.matrix.org/matrix-org/olm/-/blob/master/docs/signing.md#signing-one-time-keys